ambry-protocol/src/main/java/com.github.ambry.protocol/ReplicaMetadataResponseInfo.java

```diff
@@ -48,9 +50,9 @@ private ReplicaMetadataResponseInfo(PartitionId partitionId, FindToken findToken
     }
     this.partitionId = partitionId;
     this.remoteReplicaLagInBytes = remoteReplicaLagInBytes;
-    messageInfoListSerDe =
-        new MessageInfoListSerde(messageInfoList, getMessageInfoListVersion(replicaMetadataResponseVersion));
-    messageInfoListSize = messageInfoListSerDe.getMessageInfoListSize();
+    messageInfoAndMetadataListSerde =
+        new MessageInfoAndMetadataListSerde(messageInfoList, getMessageInfoListVersion(replicaMetadataResponseVersion));
```

Comment:
can you add a TODO here and in PartitionResponseInfo that will be calling into the other constructor

Reply:
Also, I don't find the other constructor for ReplicaMetadataResponseInfo which takes in List<MessageMetadata> . Is the plan to add it in the next patch (write) ?

Reply:
`ReplicaMetadataResponse` will always have null `MessageMetadata` as the request/response never goes to the MessageFormat.